src/top/data/bbox_reg_util.py

```diff
@@ -86,72 +89,85 @@ class CropObject(object):
 
     @dataclass
     class Settings(Serializable):
-        crop_img_size: Tuple[int, int, int] = (224, 224)
+        crop_img_size: Tuple[int, int] = (224, 224)
 
     def __init__(self, opts: Settings):
         self.opts = opts
 
+    # FIXME(Jiyon): make batch with cropped image, np,scipy -> th
     def __call__(self, inputs: dict):
         # Parse inputs
         image = inputs[Schema.IMAGE]
-        class_index = inputs[Schema.CLASS]
         num_object = inputs[Schema.INSTANCE_NUM]
-        num_keypoints = inputs[Schema.KEYPOINT_NUM]
         translation = inputs[Schema.TRANSLATION]
         orientation = inputs[Schema.ORIENTATION]
         scale = inputs[Schema.SCALE]
+        visibility = inputs[Schema.VISIBILITY]
+
+        c, h, w = image.shape[:]
+        keypoints_2d_uv = inputs[Schema.KEYPOINT_2D]
+
+        
+        keypoints_2d = th.as_tensor(keypoints_2d_uv) * th.as_tensor([w, h, 1.0])
+        num_vertices = keypoints_2d.shape[-2]
+        keypoints_2d = keypoints_2d.reshape(-1,num_vertices,3)
+
+        # clamp for the case that keypoints is in out of image
+        keypoints_2d_clamp = th.clamp(th.as_tensor(keypoints_2d_uv), min=0, max=1)
+        keypoints_2d_clamp = th.as_tensor(keypoints_2d_clamp) * th.as_tensor([w, h, 1.0])
+        keypoints_2d_clamp = keypoints_2d_clamp.reshape(-1,num_vertices,3)
+
+        orientation = th.as_tensor(orientation)
+        orientation = orientation.reshape(-1,3,3)
+        quaternions = matrix_to_quaternion(orientation)
+
+        scale = th.as_tensor(scale)
+        scale = scale.reshape(-1,3)
 
-        h, w = image.shape[-2:]
-        keypoints_2d = np.split(inputs[Schema.KEYPOINT_2D], np.array(np.cumsum(num_keypoints)))
-        keypoints_2d = [points.reshape(-1,3) for points in keypoints_2d]
-        keypoints_2d = [np.multiply(keypoint, np.array([w, h, 1.0], np.float32)).astype(int)
-                        for keypoint in keypoints_2d]
+        translation = th.as_tensor(translation)
+        translation = translation.reshape(-1,3)
+
+        point_min = th.min(keypoints_2d_clamp, dim=1).values
```

Comment:
Is `values` required here? What's the point?

Reply:
An object with `values` ​​and `indices` is returned as a result of `th.min()`. Among them, this is a code to get only `values`.

I will clarify it as `keypoint_2d` rather than `point`.